### PR TITLE
Support creating announcements via API

### LINKED ISSUE #2749: API call to set Announcement Banner
It would be nice to have an API Call to set the Icingaweb2 Announcement Banner

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #2749

### CODE REVIEW

**Comment by SimonHoenscheid** [comment]
will this be merged?

**Comment by Al2Klimov** [comment]
@SimonHoenscheid Ask @lippserd not me.

**Comment by SimonHoenscheid** [comment]
@lippserd will you merge this?

**Comment by lippserd** [comment]
@SimonHoenscheid I'll have a look at this one after the 2.5.0 release.

**Comment by SimonHoenscheid** [comment]
thanks @lippserd

File: library/Icinga/Web/Form/Element/DateTimePicker.php [hunk 1]
```diff
@@ -56,6 +56,10 @@ public function getFormat()
      */
     public function isValid($value, $context = null)
     {
+        if (is_scalar($value) && $value !== '' && ! preg_match('/\D/', $value)) {
+            $value = DateTime::createFromFormat('U', $value)->format($this->getFormat());
```
Comment (lippserd) [inline library/Icinga/Web/Form/Element/DateTimePicker.php]: This should be `(new DateTime())->setTimestamp()` I think.

File: library/Icinga/Web/Request.php [hunk 2]
```diff
@@ -118,4 +119,14 @@ public function protectId($id)
         }
         return $id . '-' . $this->uniqueId;
     }
+
+    /**
```
Comment (lippserd) [inline library/Icinga/Web/Request.php]: Please remove the inheritdoc block.

**Pull request merged**